spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -3299,6 +3385,10 @@ private Collection<ConsumerRecord<K, V>> getHighestOffsetRecords(ConsumerRecords
 					.values();
 		}
 
+		private BiConsumer<ConsumerRecord<K, V>, RuntimeException> getCallbackForAsyncFailure() {
```

Comment:
Well. I meant this contract:
```
private void callbackForAsyncFailure(ConsumerRecord<K, V> cRecord, RuntimeException ex) {
    this.failedRecords.addLast(new FailedRecordTuple<>(cRecord, ex));
}
```

Then it could be used like this:
```
adapterListener.setCallbackForAsyncFailure(this::callbackForAsyncFailure);
```